### PR TITLE
Fixing build scripts to support android build.

### PR DESCRIPTION
**Pull request by guworks**
Provides proper android gyp flag and add position-independent executable compiler flags for android builds.

### CODE REVIEW

File: common.gypi [hunk 1]
```diff
@@ -40,6 +40,10 @@
           ['OS != "win"', {
             'defines': [ 'EV_VERIFY=2' ],
           }],
+          ['OS == "android"', {
+            'cflags': [ '-g', '-O0', '-fwrapv', '-fPIE' ],
```
Comment (saghul) [inline common.gypi]: Are these needed now? Also, why the -O0?

File: android-configure [hunk 2]
```diff
@@ -16,5 +16,5 @@ export PLATFORM=android
 
 if [ $2 -a $2 == 'gyp' ]
   then
-    ./gyp_uv.py -Dtarget_arch=arm -DOS=android
+    ./gyp_uv.py -Dtarget_arch=arm -DOS=android -f make-android
```
Comment (saghul) [inline android-configure]: Why force the generator?

**Comment by saghul** [comment]
Left some comments. Also, can you please adjust your commit log per the guidelines in CONTRIBUTING.MD?

File: android-configure (hunk 2 above)
Comment (guworks) [inline android-configure]: If we don't specify the target as "android", the gyp scripts don't handle android cross compilation properly and end up trying to compile against a darwin target environment and using the wrong include files, libs, etc.

File: common.gypi (hunk 1 above)
Comment (guworks) [inline common.gypi]: We only need PIE for android. I should have avoided copying these other flags from above.

File: android-configure (hunk 2 above)
Comment (saghul) [inline android-configure]: By target I mean the -f make-android. We never had it in our script, is it
really necessary?

Comment (guworks) [inline android-configure]: Yep it needs it. This sets the gyp build flavor to android. Without it cross compilation doesn't work as it tries to use some host settings. For some more insight into where it's used see ./build/gyp/pylib/gyp/generator/make.py and gyp.common.GetFlavor(params)

Comment (saghul) [inline android-configure]: Ack.

**Comment by saghul** [comment]
LGTM. Can you please rebase?

**Comment by guworks** [comment]
Rebased!

File: common.gypi [hunk 3]
```diff
@@ -35,7 +35,16 @@
         'xcode_settings': {
           'GCC_OPTIMIZATION_LEVEL': '0',
           'OTHER_CFLAGS': [ '-Wno-strict-aliasing' ],
-        }
+        },
+        'conditions': [
+          ['OS != "win"', {
```
Comment (saghul) [inline common.gypi]: Please delete this condition, I removed it in a recent commit myself.

**Comment by guworks** [comment]
Done!

**Comment by saghul** [comment]
Landed in d57c7e6, thanks!

**Pull request closed without merging**